### PR TITLE
Updating default HAProxy config to conform to current industry standards

### PR DESCRIPTION
**Pull request by cdchris12**
# Checklist
- [X] Affected Issues have been mentioned in the Closing issues section
- [X] Documentation has been written/updated
- [X] PR title is ready for changelog and subsystem label(s) applied

 This PR updates the supported ciphers in our HAProxy config to reflect current industry standards. This commit removes TLS 1.0 and 1.1 in `intermediate` and `modern` configs, but only removes TLS 1.0 support in the `old` configuration.

Of note here is that the `modern` and `intermediate` configs are the same because the versions of HAProxy and OpenSSL used in the Openshift router image are not new enough to support the bleeding edge of tech.

### CODE REVIEW

File: openshift-setup/router-template.cfg [hunk 1]
```diff
@@ -49,35 +49,35 @@ global
...
-  ssl-default-bind-options no-sslv3
-
 # The default cipher suite can be selected from the three sets recommended by https://wiki.mozilla.org/Security/Server_Side_TLS,
 # or the user can provide one using the ROUTER_CIPHERS environment variable.
 # By default when a cipher set is not provided, intermediate is used.
 {{- if eq (env "ROUTER_CIPHERS" "intermediate") "modern" }}
   # Modern cipher suite (no legacy browser support) from https://wiki.mozilla.org/Security/Server_Side_TLS
   tune.ssl.default-dh-param 2048
-  ssl-default-bind-ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256
+  ssl-default-bind-ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384
```
Comment (dasrecht) [inline openshift-setup/router-template.cfg:58]: Can you double check those suites - they don't seem to match with what Mozilla defines under the modern configuration profile

File: openshift-setup/router-template.cfg [hunk 2]
```diff
@@ -49,35 +49,35 @@ global
...
-
+  ssl-default-bind-ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA256:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA
+  ssl-default-bind-options no-sslv3 no-tlsv10 no-tls-tickets
```
Comment (dasrecht) [inline openshift-setup/router-template.cfg]: we might want also to drop tlsv11 though we'll be not able to support tlsv13 yet due to - https://blog.qualys.com/ssllabs/2018/11/19/grade-change-for-tls-1-0-and-tls-1-1-protocols

File: openshift-setup/router-template.cfg [hunk 3]
```diff
@@ -49,35 +49,35 @@ global
...
   # Intermediate cipher suite (default) from https://wiki.mozilla.org/Security/Server_Side_TLS
   tune.ssl.default-dh-param 2048
-  ssl-default-bind-ciphers ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS
+  ssl-default-bind-ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384
+  ssl-default-bind-options no-sslv3 no-tlsv10 no-tlsv11 no-tls-tickets
```
Comment (dasrecht) [inline openshift-setup/router-template.cfg:66]: we might want also to drop tlsv11 though we'll be not able to support tlsv13 yet due to - https://blog.qualys.com/ssllabs/2018/11/19/grade-change-for-tls-1-0-and-tls-1-1-protocols

File: openshift-setup/router-template.cfg (hunk 2 above)
Comment (dasrecht) [inline openshift-setup/router-template.cfg]: old should still be `no-sslv3 no-tls-tickets`

File: openshift-setup/router-template.cfg (hunk 1 above)
Comment (cdchris12) [reply openshift-setup/router-template.cfg:58]: Our HAProxy and OpenSSL versions are too old to support the `modern` configuration : https://ssl-config.mozilla.org/#server=haproxy&version=1.8.17&config=old&openssl=1.0.2k-fips&guideline=5.4

File: openshift-setup/router-template.cfg (hunk 3 above)
Comment (cdchris12) [reply openshift-setup/router-template.cfg:66]: Already dropping tlsv11 with `no-tlsv11`

**Pull request merged**